nginx-fem-staging-redirects.conf

```diff
@@ -0,0 +1,32 @@
+set $fe_project_uri "https://fe-project.preview.zooniverse.org";
+set $fe_content_pages_uri "https://fe-content-pages.preview.zooniverse.org";
+set $fe_project_host "fe-project.preview.zooniverse.org";
+set $fe_content_pages_host "fe-content-pages.preview.zooniverse.org";
+
+# Project app data and static files
+location ~* ^/projects/(?:_next|assets)/.+?$ {
+    resolver 1.1.1.1;
+    proxy_pass $fe_project_uri;
+    proxy_set_header Host $fe_project_host;
+}
+
+# About pages data and static files
+location ~* ^/about/(?:_next|assets)/.+?$ {
+    resolver 1.1.1.1;
+    proxy_pass $fe_content_pages_uri;
+    proxy_set_header Host $fe_content_pages_host;
+}
+
+# Zooniverse About pages
+location ~* ^/about/(?:team|publications)/?$ {
+    resolver 1.1.1.1;
+    proxy_pass $fe_content_pages_uri;
+    proxy_set_header Host $fe_content_pages_host;
+}
+
+# FEM projects app: home,about and classify pages
+location ~* ^/projects/[\w-]+/[\w-]+/?(?:(classify|about)(?:/.+?)?)?/?$ {
+    resolver 1.1.1.1;
+    proxy_pass $fe_project_uri;
+    proxy_set_header Host $fe_project_host;
+}
```

Comment:
I had something similar but I split this out into two blocks for slightly more configurable values for FEM specific routing on associated project pages

- one for the `/projects/slug/name(/)`routes
- the other for the FEM specific sub pages and fallback for PFE
```
# FEM Projects app routes for project index page (optional trailing slash)
location ~* ^/projects/[\w-]*?/[\w-]+?/?$ {
    resolver 1.1.1.1;
    proxy_pass $fe_project_uri;
    proxy_set_header Host $fe_project_host;
}

# FEM Projects app routes for per classify and about pages
# but fallback to PFE when we aren't on a specific FEM project app route
# e.g. /talk, /recents, /collections etc
# note the trailing slash after slug is not options and the slug matches are lazy.
location ~* ^/projects/[\w-]*?/[\w-]+?/(?:(classify|about)(?:/.+?)?)/?$ {
    resolver 1.1.1.1;
    proxy_pass $fe_project_uri;
    proxy_set_header Host $fe_project_host;
}
```

Reply:
I'm curious about this expression: `^/projects/[\w-]*?/[\w-]+?/?$` Is that a deliberate choice to match empty owner names eg. `/projects//project-name`?
https://regexr.com/69hos

Reply:
yes - it can 404 at the FEM app vs PFE

equally it could be re-written to `^/projects/[\w-]+?/[\w-]+?/?$` to avoid this issue as you have in yours

Reply:
Aha! That makes sense. PFE would try to route it in the browser, where as Next will actively send a 404 response.

Reply:
OK, I've split those rules up into two blocks like yours.